components/prism-glimmer.js

```diff
@@ -0,0 +1,290 @@
+/* eslint-disable no-use-before-define */
+/*tri
+* Much of this implementation is copied from highlightjs-glimmer
+*
+*/
+function setup(Prism) {
+	// Merged: https://github.com/emberjs/rfcs/pull/560
+	const _EQUALITY_HELPERS = 'eq neq';
+
+	// Merged: https://github.com/emberjs/rfcs/pull/561
+	const _NUMERIC_COMPARISON_HELPERS = 'gt gte le lte';
+
+	// Merged: https://github.com/emberjs/rfcs/pull/562/files
+	const _LOGICAL_OPERATOR_HELPERS = 'and or not';
+
+	const _BLOCK_HELPERS = 'let each each-in if else unless';
+	const _DEBUG_HELPERS = 'log debugger';
+	const _INLINE_HELPERS = 'has-block concat fn component helper modifier get hash query-params';
+	const _MODIFIERS = 'action on';
+	const _SPECIAL = 'outlet yield';
+	const _LITERALS = 'true false undefined null';
+
+	function toKeywords(spaceSep) {
+		return new RegExp(`\\b(?:${spaceSep.split(' ').join('|')})\\b`);
+	}
+
+
+	// Open parenthesis for look-behind
+	const parenOpen = '(\\()';
+	const mustacheOpen = '(\\{\\{\\{?)';
+	// Symbol name. See https://www.gnu.org/software/emacs/manual/html_node/elisp/Symbol-Type.html
+	// & and : are excluded as they are usually used for special purposes
+	const symbol = '[-+*/_~!@$%^=<>{}\\w]+';
+
+	const COMPONENT_NAME_SEGMENT = /[A-Za-z0-9]+/;
+	const COMPONENT_NAME = regex.either(
+		COMPONENT_NAME_SEGMENT,
+		/[a-zA-Z0-9]+\.[a-zA-Z0-9-]+/,
+		regex.concat(COMPONENT_NAME_SEGMENT, /::/, /-?/, COMPONENT_NAME_SEGMENT),
+	);
+
+	const NUMBER = /\b0x[\dA-Fa-f]+\b|(?:\b\d+(?:\.\d*)?|\B\.\d+)(?:[Ee][+-]?\d+)?/;
+
+	const STRING =  new RegExp(regex.either(
+		/"[^{"]+/, /"/,
+		/'[^{']+/, /'/,
+		/"[^"]+"/,
+		/'[^']+'/,
+	));
+
+	const Arg = {
```

Comment:
Please use lowerCamelCase for variables.

Reply:
I think making all variables lowerCamelCase would make things harder to reason about
-- I needed more conventions:

`PascalCase` - spreadable pre-configured set up grammar-key to grammar
`SHOUT_CASE` - Regex or other simpler things

Even with this, I don't think I'm 100% consistent

Reply:
PascalCase is usually reserved from classes. The JS community is pretty consistent with that. Please don't use it for variables.

As for upper snake case/shout case, this is usually used for constant (e.g. enum constants or magic numbers). Using it for regexes a little strange but I'm not against it.